test/gurka/test_osrm_serializer.cc

```diff
@@ -146,9 +146,9 @@ TEST(Standalone, HeadingNumberCrossRoad) {
 
   auto json = gurka::convert_to_json(result, Options::Format::Options_Format_osrm);
   std::vector<int> expected_headings{1, 4, 1};
-  for (int i = 0; i < json["routes"][0]["legs"][0]["steps"].GetArray().Size(); ++i) {
-    for (int j = 0; j < json["routes"][0]["legs"][0]["steps"][i]["intersections"].GetArray().Size();
-         ++j) {
+  for (size_t i = 0; i < json["routes"][0]["legs"][0]["steps"].GetArray().Size(); ++i) {
```

Comment:
i think this is actually `unsigned` vs `size_t` which means the `operator[]` will narrow it again

https://github.com/Tencent/rapidjson/blob/master/include/rapidjson/rapidjson.h#L415